preprocessors/openstreetmap/osm_service.py

```diff
@@ -123,20 +122,27 @@ def process_streets_data(OSM_data):
                 oneway = bool(oneway)
             else:
                 oneway = oneway
-                way_object = {
-                    "street_id": int(way.id),
-                    "street_name": way.tags.get("name"),
-                    "street_type": way.tags.get("highway"),
-                    "addr:street": way.tags.get("addr:street"),
-                    "surface": way.tags.get("surface"),
-                    "oneway": oneway,
-                    "sidewalk": way.tags.get("sidewalk"),
-                    "maxspeed": way.tags.get("maxspeed"),
-                    "lanes": lanes,
-                    "nodes": node_list,
-                }
-                # Delete key if value is empty
-                way_object = dict(x for x in way_object.items() if all(x))
+            way_object = {
+                "street_id": int(way.id),
+                "street_name": way.tags.get("name"),
+                "street_type": way.tags.get("highway"),
+                "addr:street": way.tags.get("addr:street"),
+                "surface": way.tags.get("surface"),
+                "oneway": oneway,
+                "sidewalk": way.tags.get("sidewalk"),
+                "maxspeed": way.tags.get("maxspeed"),
+                "lanes": lanes,
+
+            }
+            # Fetch as many tags as possible
+
+            way_object["nodes"] = node_list
+
+            # Delete key if value is empty
+            way_object = dict(x for x in way_object.items() if all(x))
+
+            # Include only streets with names
+            if "street_name" in way_object:
```

Comment:
I'm still skeptical that we should drop all nameless paths from the output. Yes, this is useful for Florian's output, but restricts how we can use the data later on. Since this is so minor, keep it in now, but we may have to remove these lines.

Reply:
> I'm still skeptical that we should drop all nameless paths from the output. Yes, this is useful for Florian's output, but restricts how we can use the data later on. Since this is so minor, keep it in now, but we may have to remove these lines.

Ok